java/src/main/kotlin/io/spine/validation/java/generate/option/RequiredFieldGenerator.kt

```diff
@@ -72,26 +76,32 @@ internal class RequiredFieldGenerator(
             """
             if (${field.hasDefaultValue()}) {
                 var fieldPath = ${parentPath.resolve(field.name)};
-                var violation = ${violation(ReadVar("fieldPath"))};
+                var typeName =  $parentName != null ? $parentName : $TypeNameClass.of(this);
```

Comment:
As a side comment.

This thing:
```
 ... $parentName != null ? $parentName : $TypeNameClass.of(this)
```
is now a frequent guest throughout this PR.  Just use a search by `$parentName != null ?`.

Please consider extracting it somehow.